core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCreationSetupService.java

```diff
@@ -214,6 +217,12 @@ public Cluster prepare(ClusterRequest request, Stack stack, Blueprint blueprint,
         return savedCluster;
     }
 
+    private void decorateStackWithCustomDomainIfAdOrIpaJoinable(Stack stack, Cluster cluster) {
```

Comment:
unit test, please. the calling method is much larger but writing a dedicated test case which tests this case would be more than nothing.